packages/react-graphql/src/hooks/useQuery.ts

```diff
@@ -0,0 +1,202 @@
+import ApolloClient, {
+  OperationVariables,
+  ObservableQuery,
+  ApolloError,
+  WatchQueryOptions,
+} from 'apollo-client';
+import {ObservableQueryFields} from 'react-apollo';
+import {DocumentNode} from 'graphql-typed';
+import {useEffect, useMemo, useState} from 'react';
+
+import {AsyncQueryComponentType} from '@shopify/react-graphql';
+
+import {
+  QueryHookOptions,
+  QueryHookResult,
+  QueryDocumentResolving,
+} from './types';
+import {useApolloClient} from './ApolloContext';
+import useGraphQLDocument from './useGraphQLDocument';
+import {
+  getCachedObservableQuery,
+  invalidateCachedObservableQuery,
+} from './queryCache';
+import {compact, objectToKey, actHack} from './utilities';
+
+export default function useQuery<Data = any, Variables = OperationVariables>(
```

Comment:
_carry over things that's not resolved yet but lost due to moving files_

https://github.com/Shopify/quilt/pull/642#discussion_r273720608
> Do you need overrides for this? I would have assumed you'd be OK without them since the thing you pass the queryOrComponent to accepts both types?

> Also, random, but I've been getting in the habit of only using named exports. Want to join the club? :P

Reply:
I mostly did the overriding so that if you pass in a queryDocument, you won't need to check to see if all the helper methods (like refetch), since I can infer the return type from what's being pass in.